src/osemgrep/cli_ci/Ci_subcommand.ml

```diff
@@ -336,6 +336,10 @@ let finding_is_blocking (m : OutJ.cli_match) =
 
   match metadata with
   | JSON.Object xs ->
+      let should_check_validation_state =
+        Option.is_some m.extra.validation_state
+      in
```

Comment:
Seems odd to have this given we check this same condition 3 lines later. Seems like this whole chunk could be re-written as
```ocaml
  match metadata with
  | JSON.Object xs ->
        (match
          ( m.extra.validation_state,
            List.assoc_opt "dev.semgrep.validation_state.actions" xs,
            List.assoc_opt "dev.semgrep.actions" xs )
        with
        | Some validation_state, Some (JSON.Object vs) ->
            List.assoc_opt (validation_state_to_action validation_state) vs
            |> Option.map (JSON.equal (JSON.String "block"))
            |> Option.value ~default:false
        | None, Some (JSON.Array actions) -> contains_blocking actions
        | _ -> false)
  | _ -> false
```

Reply:
Much cleaner. Done in [1a3a69e](https://github.com/semgrep/semgrep/pull/9943/commits/1a3a69e442e8274e7bbcc565baa4cc3c86e88c64)